src/lightning/fabric/fabric.py

```diff
@@ -653,16 +657,26 @@ def launch(self, function: Optional[Callable[["Fabric"], Any]] = None, *args: An
                 "This script was launched through the CLI, and processes have already been created. Calling "
                 " `.launch()` again is not allowed."
             )
-        if function is not None and not inspect.signature(function).parameters:
+        if function is not _do_nothing:
+            if not callable(function):
+                raise TypeError(
+                    f"`Fabric.launch(...)` needs to be a callable. Given {function}."
+                    " HINT: do `.launch(your_fn)` instead of .launch(your_fn())"
+                )
+            if not inspect.signature(function).parameters:
+                raise TypeError(
+                    f"`Fabric.launch(function={function})` needs to take at least one argument. The launcher will"
+                    " pass in the `Fabric` object so you can use it inside the function."
+                )
+        elif isinstance(self.strategy, XLAStrategy):
```

Comment:
If you do this, then you could also add a similar error message in the case where strategy.launcher is the MultiProcessingLauncher.